nwbinspector/checks/time_series.py

```diff
@@ -99,15 +99,17 @@ def check_resolution(time_series: TimeSeries):
 def check_unit_formatting(time_series: TimeSeries):
     """
     Check the unit value of a TimeSeries that it complies with CMIXF-12 convention for formatting the units.
-    
+
     Best Practice: :ref:`best_practice_units_of_measurement`
     """
 
     # Early return for arbitrary units that are unknown or unavailable.
     if time_series.unit == "a.u.":
         return
 
+    lexer = CMIXFLexer()
+    tokens = lexer.tokenize(time_series.unit)
```

Comment:
`lexer.tokenize()` a `generator` object, when a bad character is found it immediately raises an error. e.g.

```
tokens = lexer.tokenize("ta")
next(tokens) 
```
would return the recognized token value for unit t
`Token(type='UNITC', value='t', lineno=1, index=0)`

next(tokens)
would return:
`{ValueError}Line 1: Bad character 'a'`